File: src/main/java/BSCLM/SongCellFactory.java
package BSCLM;

import javafx.scene.control.ListCell;
import javafx.scene.control.ListView;
import javafx.util.Callback;

public class SongCellFactory implements Callback<ListView<Song>, ListCell<Song>> {
    @Override
    public ListCell<Song> call(ListView<Song> param) {
        return new ListCell<Song>(){
            @Override
            public void updateItem(Song song, boolean empty) {
                super.updateItem(song, empty);
                if (empty || song == null) {
                    setText(null);
                } else {
                    String songName = song.getSongName();
                    if (songName.equals(""))
                        songName = "Untitled";
                    String authorName = song.getSongAuthorName();
                    if (authorName.equals(""))
                        authorName = "Noname";
                    setText(songName + "-" + authorName);
                }
            }
        };
    }
}

File: src/main/java/BSCLM/SongCellFactory.kt
package BSCLM

import javafx.scene.control.ListCell
import javafx.scene.control.ListView
import javafx.util.Callback

class SongCellFactory: Callback<ListView<Song>, ListCell<Song>> {
    override fun call(param: ListView<Song>?): ListCell<Song> {
        return object : ListCell<Song>() {
            override fun updateItem(song: Song?, empty: Boolean) {
                super.updateItem(song, empty)
                text = if (empty || song == null) {
                    null
                } else {
                    "${song.getSongName().ifEmpty { "Untitled" }}-${song.getSongAuthorName().ifEmpty { "Noname" }}"
                }
            }
        }
    }
}